src/runtime/types.cpp

```diff
@@ -304,6 +304,21 @@ extern "C" BoxedFunctionBase::BoxedFunctionBase(CLFunction* f, std::initializer_
     assert(f->num_defaults == ndefaults);
 }
 
+BoxedFunction::BoxedFunction(CLFunction* f) : BoxedFunction(f, {}) {
+}
+
+BoxedFunction::BoxedFunction(CLFunction* f, std::initializer_list<Box*> defaults, BoxedClosure* closure,
+                             bool isGenerator)
+    : BoxedFunctionBase(f, defaults, closure, isGenerator) {
+
+    // TODO eventually we want this to assert(f->source), I think, but there are still
+    // some builtin functions that are BoxedFunctions but really ought to be a type that
+    // we don't have yet.
+    if (f->source) {
+        this->name = static_cast<BoxedString*>(boxString(f->source->getName()));
```

Comment:
does `name` get a default value if this doesn't get hit?  I think we may have to explicitly initialize it to NULL.